File: FinalProject/app/src/main/java/com/example/finalproject/items/Recipe.java
package com.example.finalproject.items;

import android.util.Pair;

import androidx.recyclerview.widget.RecyclerView;

import com.example.finalproject.items.Item;

import java.util.ArrayList;

public class Recipe {
    private Item product;
    private ArrayList<Pair<Item, Integer>> ingredients=new ArrayList<>();

    public Recipe(Item product, ArrayList<Pair<Item, Integer>> ingredients) {
        this.product = product;
        this.ingredients.addAll(ingredients);
    }

    public Item getProduct() {
        return product;
    }

    public void setProduct(Item product) {
        this.product = product;
    }

    public ArrayList<Pair<Item, Integer>> getIngredients() {
        return ingredients;
    }

    public void setIngredients(ArrayList<Pair<Item, Integer>> ingredients) {
        this.ingredients = ingredients;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/items/Recipe.kt
package com.example.finalproject.items

import com.example.finalproject.entities.Player

class Recipe(val product:Pair<Item, Int>, private val ingredients:ArrayList<Pair<Item, Int>>){

    init {
        this.ingredients.clear()
        this.ingredients.addAll(ingredients)
    }

    @JvmName("getIngredients1")
    fun getIngredients():ArrayList<Pair<Item, Int>>{
        return ArrayList(ingredients)
    }

    @JvmName("getProduct1")
    fun getProduct():Pair<Item, Int>{
        return Pair(product.first.clone(), product.second)
    }
}